ui_framework/test/common_tests.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render } from 'enzyme';
+
+/**
+ * Ensures the rendered class matches the jest snapshot.
+ * @param reactClass
+ * @param props
+ */
+export function basicRenderTest(reactClass, props = {}) {
```

Comment:
In might just be a case of different philosophies.

Maybe the problem is trying to "recreate" the normal view of unit tests — that you have to test everything in separate tests. I'd probably write these tests as:

```js
import { render, basicHtmlAttributes } from '../../test/common_tests';

import {
  KuiToolBar,
  KuiToolBarFooter,
} from './tool_bar';

test('renders KuiToolBar', () => {
  const component = <KuiToolBar { ...basicHtmlAttributes }>children</KuiToolBar>
  expect(render(component)).toMatchSnapshot();
})

test('renders KuiToolBarFooter', () => {
  const component = <KuiToolBar { ...basicHtmlAttributes }>children</KuiToolBar>
  expect(render(component)).toMatchSnapshot();
})
```

And for the table stuff I'd probably create a couple of snapshot tests. Though, I'm probably more of the "pragmatic type".